### PR TITLE
fix: gatekeeper postinstall job securityContext

### LINKED ISSUE #688: The gatekeeper post install job cannot run pod due to PSP and insufficient securityContext
**Describe the bug:** (a clear and concise description of what the bug is)

**To Reproduce**

1. Deploy otomi with minimal values set
2. Observe CreateContainerConfigError at postinstall Pod
```
Error: container has runAsNonRoot and image will run as root (pod: "gatekeeper-postinstall-job-h7l8r_default(2222df8f-1252-47c3-8b0c-15875010de94)", container: postinstall)
```

### PR DESCRIPTION
**Pull request by j-zimnowoda**
fixes #688

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file.

### CODE REVIEW

File: charts/gatekeeper-operator/templates/post-job.yaml [hunk 1]
```diff
@@ -43,5 +43,6 @@ spec:
               memory: 16Mi
           securityContext:
             runAsNonRoot: true
+            runAsUser: 65534
```
Comment (Morriz) [inline charts/gatekeeper-operator/templates/post-job.yaml:46]: why is this needed? we chose to use `runAsNonRoot` which provides the same security as choosing a non-root UID. I recommend to close this PR

Comment (Morriz) [reply charts/gatekeeper-operator/templates/post-job.yaml:46]: we should remove anything related to real `PodSecurityPolicy` resources asap.

Comment (Morriz) [reply charts/gatekeeper-operator/templates/post-job.yaml:46]: I also want to remove the ability to be able to set the UID globally via policy at all, as it serves no use case

Comment (j-zimnowoda) [reply charts/gatekeeper-operator/templates/post-job.yaml:46]: well, PSP are in 3rd party charts

Comment (j-zimnowoda) [reply charts/gatekeeper-operator/templates/post-job.yaml:46]: > we should remove anything related to real `PodSecurityPolicy` resources asap.

I do not consider it as a "hot issue", please create a user story and put it to the backlog

Comment (Morriz) [reply charts/gatekeeper-operator/templates/post-job.yaml:46]: I just read your bug report and did not know this image was using root. (Why would it?)

So yes, if the uid can be set then great

**Review by Morriz — APPROVED** [review]
Just drop the `mustRunAsNonRoot: true` line then as it is unneeded and you may merge this into master
